client.go

```diff
@@ -345,6 +340,48 @@ func (uc *Client) isEnabled(feature string, options ...FeatureOption) api.Strate
 	}
 }
 
+func (uc *Client) isParentDependencySatisfied(feature *api.Feature, context context.Context) bool {
```

Comment:
```
func (uc *Client) isParentDependencySatisfied(feature *api.Feature, context context.Context) bool {
    if feature == nil || feature.Dependencies == nil || len(*feature.Dependencies) == 0 {
        return true
    }

    for _, parent := range *feature.Dependencies {
        parentToggle := uc.repository.getToggle(parent.Feature)
        isEnabled := uc.isEnabled(parent.Feature, WithContext(context)).Enabled

        if parentToggle == nil || 
           (parentToggle.Dependencies != nil && len(*parentToggle.Dependencies) > 0) || 
           (parent.Enabled != nil && !isEnabled) {
            return false
        }

        if parent.Enabled == nil {
            variantName := uc.getVariantWithoutMetrics(parent.Feature, WithVariantContext(context)).Name
            hasValidVariant := parent.Variants != nil && len(*parent.Variants) > 0 && contains(*parent.Variants, variantName)
            
            if !hasValidVariant && !isEnabled {
                return false
            }
        }
    }

    return true
}
```

Reply:
Unless we write an every helper, we can't return out here or we will exit the entire function without properly looping over all the dependencies.